File: Java/Niveau1/7- Conditions avancées, opérateurs booléens/Maison de l'espion.java
import algorea.Scanner;

class Main{

    static Scanner entrée = new Scanner(System.in);
    
    public static void main(String[] args){
        int xMin = entrée.nextInt();
        int xMax = entrée.nextInt();
        int yMin = entrée.nextInt();
        int yMax = entrée.nextInt();
        int nombresMaison = entrée.nextInt();
        int maisonsSuspectes = 0;
        for(int i = 1; i <= nombresMaison; i++){
            int xMaison = entrée.nextInt();
            int yMaison = entrée.nextInt();

            if(((xMin <= xMaison) && (xMaison <= xMax)) && ((yMin <= yMaison) && (yMaison <= yMax))){
                maisonsSuspectes++;
            }
        }
        System.out.println(maisonsSuspectes);
    }
    
}

File: C++/Niveau 1/7 - Conditions avancées, opérateurs booléens/Maison de l'espion.cpp
#include <iostream>
using namespace std;

int main(){
    int maisonsSuspectes = 0,nombreMaisons, xMin, xMax, yMin, yMax;

    cin >> xMin >> xMax >> yMin >> yMax >> nombreMaisons;

    for(int i = 1; i <= nombreMaisons; i++){

        int x,y;

        cin >> x >> y;

        if(((xMin <= x) && (xMax >= x)) && ((yMin <= y) && (yMax >= y))){
            maisonsSuspectes++;
        }
    }

    cout << maisonsSuspectes << endl;

}